sync/index.test.ts

```diff
@@ -226,6 +226,81 @@ test('maps input actions', async () => {
   equal(pair.rightNode.log.actions(), [{ type: 'a1' }])
 })
 
+test('handles error in onActions', async () => {
+  let error = new Error('test')
+  let catched: Error[] = []
+
+  let pair = await createTest()
+  pair.rightNode.options.onActions = () => {
+    throw error
+  }
+  pair.rightNode.catch(e => {
+    catched.push(e)
+  })
+  pair.leftNode.log.add({ type: 'a' })
+
+  await delay(50)
+  equal(catched, [error])
+})
+
+test('actions are added to the log if onActions is present and if actions were added to the log inside it', async () => {
+  let pair = await createTest(created => {
+    created.rightNode.options.inFilter = async (action, meta) => {
+      type(meta.id, 'string')
+      type(meta.time, 'number')
+      return action.type !== 'c1'
+    }
+    created.rightNode.options.inMap = async (action, meta) => {
+      type(meta.id, 'string')
+      type(meta.time, 'number')
+      return [{ type: action.type + '1' }, meta]
+    }
+    created.rightNode.options.onActions = (action, meta) => {
+      created.rightNode.log.add(action, meta)
```

Comment:
This test will not find that `onActions` is not calling (because `log.add` is default behavior)

Reply:
For instance, let’s put actions to an array and then check the array that it contains actions and log is empty.